### PR TITLE
Feat department field

### LINKED ISSUE #90: Add and implement Department field
**Is necessary for #89**

The Department field will have to:

- Be in the `Person` class
- Be available to set in `add` and `edit` commands
- Be restricted to a list of customised valid departments for the company
- Be customisable in some way (can add, remove, edit)

### PR DESCRIPTION
**Pull request by kevinchangjk**
**Resolves #90**

Added a `Department` class and field for the `Person` class. Any entry for department will be checked against a list of valid departments (which for now, is fixed). 

Updated current test cases to fit this addition as well, but have not added additional test cases.

### CODE REVIEW

File: src/main/java/coydir/model/person/Department.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import static coydir.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Person's department in the database.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDepartment(String)}
+ */
+public class Department {
+
+    public static final String[] VALID_DEPARTMENTS = {
```
Comment (chao890) [inline src/main/java/coydir/model/person/Department.java:12]: Comprehensive list of departments, good job!

File: src/main/java/coydir/model/person/Person.java [hunk 2]
```diff
@@ -33,13 +34,14 @@ public class Person {
      * Every field must be present and not null.
      */
     public Person(Name name, EmployeeId employeeId, Phone phone, Email email, Position position,
-            Address address, Set<Tag> tags, int numberOfLeaves) {
-        requireAllNonNull(name, employeeId, position);
+            Department department, Address address, Set<Tag> tags, int numberOfLeaves) {
+        requireAllNonNull(name, employeeId, phone, email, position, department, address);
```
Comment (chao890) [inline src/main/java/coydir/model/person/Person.java:38]: Just to clarify, only numberOfLeaves is not required to initialize an employee?

**Comment by chao890** [comment]
Very Nice.

File: src/main/java/coydir/model/person/Person.java (hunk 2 above)
Comment (prit3010) [reply src/main/java/coydir/model/person/Person.java:38]: Actually why are the rest Non Null now?

**Review by prit3010 — APPROVED** [review]
Looks Good. Just wanted to ask about the user adding their own departments? Is that something we will be exploring?

File: src/main/java/coydir/model/person/Person.java (hunk 2 above)
Comment (kevinchangjk) [reply src/main/java/coydir/model/person/Person.java:38]: Always has been. This is for `Person` class, so the related fields must all be present, just that the optional ones can be the `NULL` objects instead. But the objects themselves are still there.

Comment (kevinchangjk) [reply src/main/java/coydir/model/person/Person.java:38]: > Just to clarify, only numberOfLeaves is not required to initialize an employee?

Actually I suppose it should also be not null. I can go add that in.

**Comment by kevinchangjk** [comment]
> Looks Good. Just wanted to ask about the user adding their own departments? Is that something we will be exploring?

Yeah, unfortunately not so soon I think, because from what I envision now, that comes with either:

1. Adding 2-3 commands in order to view/add/delete departments
2. Storing departments in a separate file and asking users to edit them directly

Both these options are a little more complex and will take some time to implement.

**Pull request merged**